### ISSUE TITLE
Handling of "undersea" values does not properly handle land at sea level (elevation == 0m)

### ISSUE THREAD

**Issue by DFEvans** [bunburya/bother#5]
The undersea value handling works well for negative elevations. However, for land at exactly sea level (input value 0), the modification is not applied, leading to land below 0m being rendered, land above 0m being rendered, but land at exactly 0m is considered to be sea level.

The input SRTM data does distinguish between them - a value of 0 indicates land, but at sea level, whereas nodata indicates water itself. The tool loses this distinction quite early on in `create_tif_file`, where the various SRTM rasters are merged and any nodata areas are set to zero.

It would instead be better to use another nodata value (-255? -32768? NaN can't be used without casting the data to float), and to ignore this through most of the processing, before finally mapping nodata to 0m on output.

The image below shows an example area along the north coast of Germany. The "undersea" areas have been pulled up, giving the right coastline, but there's many little holes for pixels at exactly 0m elevation.

![image](https://user-images.githubusercontent.com/28623784/92326744-3d9b7380-f04c-11ea-9ed1-c28b976bc819.png)

**Comment by bunburya** [comment]
Good spot - I was wondering why I was getting those "dimples" occasionally.  Can you send me the coordinates you've been using for that image so I can test out a solution?

In terms of a NODATA value to use, I think the SRTM data uses -32768 by default so I don't see an issue sticking to that.  Apparently the lowest land in the world is approx. -413m so I guess any value below that would work!

**Comment by DFEvans** [comment]
The command I've been using is:

```
bother --bounds 47.5 5.0 54.7 14.99 --epsg 4326 --raise-low 0 --raise-undersea 1 germany_4326.png
```

Just about all of the northern coast shows the issue - especially the NW part.

> Apparently the lowest land in the world is approx. -413m so I guess any value below that would work!

I suspected -255 might be too 'high' to avoid causing problems somewhere!

**Comment by bunburya** [comment]
I've pushed a commit to the handle_nodata branch that should fix this.  Preview:
![Unnamed, 21st Dec 1952](https://user-images.githubusercontent.com/531559/92332614-26be4680-f077-11ea-8589-39c5a0da3867.png)
I've refactored the code to treat NODATA (-32768) and 0 as distinct, until the to_png function where NODATA pixels are set to 0.  I've also changed the raise_undersea_land function to work on pixels with elevation <= 0 (I guess this makes the name of the function slightly inaccurate but I'm not to worried about that...)

I had to re-generate some of the TIF files in the test/examples dir as obviously these are different now (preserving the distinction between NODATA and 0).  I also had to re-generate the PNGs for Ireland because there were some minor differences between the old PNG files and the images being generated from the new TIF files - I am guessing this is because the resampling process (which uses bilinear interpolation to calculate the values of the new pixels) previously ignored all values of 0 in the interpolation process whereas it will now take account of 0 values (as opposed to NODATA values).

Let me know if this works for you - I won't merge the branch into master just yet.
